main.go

```diff
@@ -447,17 +450,43 @@ func setupControllers(ctx context.Context, mgr ctrl.Manager, sw *watch.Controlle
 
 	// Setup all Controllers
 	setupLog.Info("setting up controllers")
-	watchSet := watch.NewSet()
+
+	// Events ch will be used to receive events from dynamic watches registered
+	// via the registrar below.
+	events := make(chan event.GenericEvent, 1024)
+	w, err := wm.NewRegistrar(
```

Comment:
nit: `w := thing that begins with R`  is a bit confusing.  Could it be `r`, `reg`, `wmReg`?

Reply:
good call! this was probably c/p'd; I changed it all over the files I touched now